app/src/main/java/com/gh4a/activities/CommitActivity.java

```diff
@@ -237,4 +237,15 @@ private void loadComments(boolean force) {
                     showContentIfReady();
                 }, this::handleLoadFailure);
     }
+
+    @Nullable
+    @Override
+    protected Uri getActivityUri() {
+        if (mPullRequestNumber > 0)
+        return Uri.parse("https://github.com/" + mRepoOwner + "/" + mRepoName + "pull/" +
```

Comment:
Suggestion: Start with `IntentUtils.createBaseUriForRepo()` and `append()` to that. See `onOptionsItemSelected` for an example. Doing so also allows sharing most of the generation in the two branches:
```
Uri.Builder builder = IntentUtils.createBaseUriForRepo(mRepoOwner, mRepoName);
if (mPullRequestNumber > 0) {
    builder.appendPath("pull")
            .appendPath(String.valueOf(mPullRequestNumber));
}
return builder.appendPath("commit")
        .appendPath(mObjectSha)
        .build();
```

Reply:
Slight spin on what you've suggested but the base uri methods and Uri.Builder approach look better.

Reply:
Why return the `Uri.Builder` though? The activity doesn't intend to modify the URI after all.